### DISCUSSION TITLE
Building sbt artifacts from source

### DISCUSSION THREAD

**Discussion by inglor** [sbt/sbt#7460] [General]
Hello,

I'm trying to package `sbt` for Arch Linux and we'd like to build from source the artifacts released in Releases. When i tried the suggestion from docs [1] it doesn't seem to produce artifacts like `tgz` or `zip`.

Can you suggest how we can reproduce those artifacts locally?

Thanks,

[1]: https://github.com/sbt/sbt/blob/1.9.x/DEVELOPING.md#instruction-to-build-just-sbt

**Comment by SethTisue** [comment]
I think that stuff is in https://github.com/sbt/sbt-dist ?

**Reply by eed3si9n** [reply]
There's a shell script in the repo, which does
```bash
buildLinux() {
  pushd sbt
  cd launcher-package
  sbt -Dsbt.build.version=$SBT_VER -Dsbt.build.offline=false clean universal:packageBin universal:packageZipTarball
  mv target/universal/sbt.zip target/universal/sbt-$SBT_VER.zip
  mv target/universal/sbt.tgz target/universal/sbt-$SBT_VER.tgz
  popd
}
```

the actual build etc are in sbt/sbt, except this would basically download things from Maven Central, so technically it's not "building from source" either?

**Reply by inglor** [reply]
You are thinking the debian way maybe? - this works thought for us.

Thanks!
